frontend/rust-lib/flowy-sqlite/src/search.rs

```diff
@@ -0,0 +1,298 @@
+      view_id: view_id.to_owned(),
+      id: view_id.to_owned(),
+      data: name.to_owned(),
+    }
+  }
+
+  pub fn new_document(view_id: &str, page_id: &str, text: &str) -> Self {
+    Self {
+      index_type: IndexType::Document.to_string(),
+      view_id: view_id.to_owned(),
+      id: page_id.to_owned(),
+      data: text.to_owned(),
+    }
+  }
+}
+
+/// Add search data for searching.
+pub fn add(conn: &mut SqliteConnection, data: &SearchData) -> QueryResult<usize> {
+  sql_query("INSERT INTO search_index (index_type, view_id, id, data) VALUES (?,?,?,?)")
+    .bind::<Text, _>(&data.index_type)
+    .bind::<Text, _>(&data.view_id)
+    .bind::<Text, _>(&data.id)
+    .bind::<Text, _>(&data.data)
+    .execute(conn)
+}
+
+/// Update view name.
+pub fn update_view(conn: &mut SqliteConnection, data: &SearchData) -> QueryResult<usize> {
+  sql_query("UPDATE search_index SET data=? WHERE index_type=? and view_id=?")
+    .bind::<Text, _>(&data.data)
+    .bind::<Text, _>(IndexType::View.to_string())
+    .bind::<Text, _>(&data.view_id)
+    .execute(conn)
+}
+
+/// Search index for matches.
+pub fn search_index(conn: &mut SqliteConnection, s: &str) -> QueryResult<Vec<SearchData>> {
+  sql_query("SELECT index_type, view_id, id, data FROM search_index WHERE search_index MATCH ?")
+    .bind::<Text, _>(s)
+    .load(conn)
+}
+
+/// Delete view and data associated with the view.
+///
+/// - Document: delete the row that contains the text.
+pub fn delete_view(conn: &mut SqliteConnection, id: &str) -> QueryResult<usize> {
+  sql_query("DELETE FROM search_index WHERE view_id=?")
+    .bind::<Text, _>(id)
+    .execute(conn)
+}
+
+/// Update document text.
+pub fn update_document(conn: &mut SqliteConnection, data: &SearchData) -> QueryResult<usize> {
+  sql_query("UPDATE search_index SET data=? WHERE index_type=? and id=?")
+    .bind::<Text, _>(&data.data)
+    .bind::<Text, _>(IndexType::Document.to_string())
+    .bind::<Text, _>(&data.id)
+    .execute(conn)
+}
+
+/// Delete document data using page_id.
+pub fn delete_document(conn: &mut SqliteConnection, id: &str) -> QueryResult<usize> {
+  sql_query("DELETE FROM search_index WHERE index_type=? and id=?")
+    .bind::<Text, _>(IndexType::Document.to_string())
+    .bind::<Text, _>(id)
+    .execute(conn)
+}
+
+#[cfg(test)]
+mod tests {
+
+  use diesel_migrations::MigrationHarness;
+  use tempfile::TempDir;
+
+  use crate::{prelude::PoolConfig, Database, DB_NAME, MIGRATIONS};
+
+  use super::*;
+
+  fn setup_db() -> (TempDir, Database) {
+    let tempdir = TempDir::new().unwrap();
+    let path = tempdir.path().to_str().unwrap();
+    let pool_config = PoolConfig::default();
+    let database = Database::new(path, DB_NAME, pool_config).unwrap();
+    let mut conn = database.get_connection().unwrap();
+    (*conn).run_pending_migrations(MIGRATIONS).unwrap();
+    run_migrations(&mut conn).unwrap();
+
+    (tempdir, database)
+  }
+
+  #[test]
+  fn test_migration() {
```

Comment:
the database is always new. Does it necessary to test it?

Reply:
This test was used as part of the dev process. It's not needed per se, so I have removed it.